hack/run-hive-locally.sh

```diff
@@ -0,0 +1,296 @@
+        )
+        env "${final_env_values[@]}" make run 2>&1 |
+        tee "./$artifacts_folder/${hive_component_controllers}-logs-${hc_spec_generation}.txt" &
+
+        # To simulate the (in-cluster) functioning of Hive, hive-controllers
+        # (running locally) are re-deployed once HiveConfig.spec changes
+        wait_until_hiveconfig_spec_changes
+        hl_echo "Redeploying $component"
+        kill_background_jobs
+    done
+}
+
+function restore_and_cleanup {
+    hl_echo "Cleaning up before exit"
+
+    # Clear trap
+    trap '' EXIT SIGINT
+
+    # Exit early if component is not selected yet
+    if [[ -z "${component:+"${component}"}" ]]; then
+        hl_echo $'\nComponent not yet selected'
+        exit 0
+    fi
+
+    # Restore Hive
+    case "${component}" in
+    "$hive_component_operator")
+        hl_echo "Scaling up Deployment/hive-operator"
+        oc scale -n "$hive_operator_ns" deploy/hive-operator --replicas="$hive_operator_replicas"
+        ;;
+    "$hive_component_controllers")
+        kill_background_jobs
+        hl_echo "Disabling maintenance mode"
+        oc patch hiveconfig hive -p '{"spec": {"maintenanceMode": false}}' --type merge
+        ;;
+    *)
+        hl_echo "Invalid component $component" >&2
+        ;;
+    esac
+
+    # Clean up artifacts
+    if [[ -d "./$artifacts_folder" ]] && [[ "$cleanup_artifacts" = "true" ]]; then
+        hl_echo "Cleaning up artifacts"
+        rm -r "$artifacts_folder"
+    fi
+
+    # Propagate the exit signal to stop script execution
+    exit 0
+}
+
+# Configuration
+hive_component_operator="hive-operator"
+hive_component_controllers="hive-controllers"
+hive_components_to_choose_from=("$hive_component_operator" "$hive_component_controllers")
+hive_operator_ns="hive"
+hive_controllers_ns="$(oc get hiveconfig hive -o jsonpath='{.spec.targetNamespace}')"
+hive_controllers_ns="${hive_controllers_log_level:-hive}"
+hive_operator_replicas="$(oc get deploy "$hive_component_operator" -n "$hive_operator_ns" -o jsonpath='{.spec.replicas}')"
+hive_operator_log_level="debug"
+hive_controllers_log_level="debug"
+artifacts_folder="artifacts"
+cleanup_artifacts="false"
+hc_spec_generation=0
+kill_job_grace_period_seconds=3
+
+# Parse long options
+while [ $# -gt 0 ]; do
+    case "$1" in
+    "--hive-operator-ns="*)
+        hive_operator_ns="${1#*=}"
+        ;;
+    "--hive_operator_log_level="*)
+        hive_operator_log_level="${1#*=}"
+        ;;
+    "--hive_controllers_log_level="*)
+        hive_controllers_log_level="${1#*=}"
+        ;;
+    "--artifacts_folder="*)
+        artifacts_folder="${1#*=}"
+        ;;
+    "--cleanup_artifacts")
+        cleanup_artifacts="true"
+        ;;
+    "--help")
+        usage
+        exit 0
+        ;;
+    *)
+        hl_echo "Invalid option: $1" >&2
+        usage
+        exit 1
+        ;;
+    esac
+    shift
+done
+
+# Check if artifacts folder exists. Create it if not.
```

Comment:
IMHO there should always be a way to run non-interactively. I would maybe not even bother being nice here -- just fail if the artifacts folder exists, and let the user decide whether to empty, remove, or rename it.

Reply:
Okay, current version is non-interactive, so no need for input.